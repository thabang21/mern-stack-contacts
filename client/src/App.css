
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
 }

body {
  font-family: Arial;
  background-color: whitesmoke; 
}

/* Navbar */
.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
  z-index: 1;
  width: 100%;
  color:white;
  background-color: teal;
  margin-bottom: 70px;
}

.navbar ul {
  display: flex;
 }
 .navbar i, h1 {
  padding: 0px
}
li{
  list-style: none;
  padding: 10px;
}
a{
  text-decoration: none;
  color:white
}

/* app */
.app-container{
  max-width: 90rem;
  margin: auto;
  }
 
.main {
  display: flex;
  justify-content: space-around;
  align-items: flex-end

}

/* GuestForm */

form {
  display: flex;
  flex-direction: column;
  position: relative;
   }
   .invite-section {
     flex-basis:100%;
   }
 .invite-section h1 {
   color:teal;
   padding: 0;
   margin:1rem 0
 }
 form input[type=text] , form input[type=email] , form input[type=password]{
   border:none;
   height: 2.5rem;
  font-size:1.2rem;
  border-bottom: solid 1px gray;
 }
 form input[type=text]:focus , form input[type=email]:focus , form input[type=password]:focus{
  outline: none !important;   
}

.btn {
  display: inline-block;
  background: teal;
  color:white;
  padding: 10px;
  font-size: 1rem;

  }
    
/* guest card */
.guest-card {
 color:white;
 background:powderblue;
 margin: 0.7rem 0;
 position: relative;
 max-width: 26rem;
 line-height: 1.6;
 margin-right:1rem;
 max-width: 100%;
}
.card-head {
 display: flex;
 justify-content: space-between;
 align-items: baseline;
 background: #ccc;
 padding: 0.2rem
}
.card-head input[type="checkbox"]{
 display: none
 }
.card-head i, .card-head label {
 color:rgb(150,150, 150);
 padding: 0.1rem;
 cursor: pointer;
}

.card-head label {
  margin-right: 0.2rem
}
.card-head button {
border:none;
padding:0.4rem;
cursor: pointer;
}
.card-head button:focus{
 outline: none
}
 .card-body{
  padding:1.2rem;
  position: relative;
  display: flex;
  flex-direction: column
}
.card-body h2 {
 margin-top:1rem;
 padding: 0
}

.contact {
  display: flex;
  align-items: center
      }
 .contact p {
   font-weight: bold;
   padding: 0 0.5rem
 }

/*  register and login */
.register , .login {
  text-align: center;
  width:45%;
  margin: 12rem auto;
  }
  .clear {
   position: absolute;
   display: inline;
   background: rgb(189, 187, 187);
   bottom:0;
   right: 0;
  }
  .no-guest {
    text-align: center;
    padding-right: 6rem;
    margin-top: 3rem
  }
  .question {
    color: #333;
    margin: 1.5rem 0
  }
  .question a {
    color: teal;
      }
  .danger {
    background-color: rgb(207, 58, 58);
    width: 100%;
    margin-bottom: 0.3rem;
    margin-top:0rem;
    text-transform: capitalize;
    padding:0.5rem;
    cursor:initial;
    border:none;
    color:#fff;
    font:inherit
    
    }
  .danger span {
    float: right;
    cursor: pointer;
  }
