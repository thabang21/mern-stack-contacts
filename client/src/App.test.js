import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

test("Snapshot test", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
