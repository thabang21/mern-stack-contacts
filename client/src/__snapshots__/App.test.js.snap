// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<div>
  <div
    className="navbar"
  >
    <div
      className="logo"
    >
      <h1>
        CONTACTS BOOK 
      </h1>
      <p>
        Consider your contacts saved 
        <span>
          ❤
        </span>
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <span
        className="sm-hide"
      >
        |
      </span>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </div>
  <div
    className="app-container"
  >
    <div
      className="main"
    >
      <div
        className="filter"
      >
        <div
          className="toggle"
        >
          <label
            class="switch"
          >
            <input
              onChange={[Function]}
              type="checkbox"
            />
            <span
              class="slider round"
            />
          </label>
          <p
            className="lead"
          >
            Show favourite Contacts!
          </p>
        </div>
        <div>
          <input
            className="search"
            onChange={[Function]}
            placeholder="Search Contact by name..."
            type="text"
          />
          <i
            className="fas fa-search search-icon"
          />
        </div>
      </div>
      <div
        className="invite-section"
      >
        <h1>
          Create Contact
        </h1>
        <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={true}
            type="text"
            value=""
          />
          <input
            name="phone"
            onChange={[Function]}
            placeholder="Phone"
            required={true}
            type="text"
            value=""
          />
          <p
            className="options-label"
          >
            Contact Type
          </p>
          <div
            className="options"
          >
            <label
              class="container"
            >
              family
              <input
                checked={true}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="family"
              />
              <span
                class="checkmark"
              />
            </label>
            <label
              class="container"
            >
              Friends
              <input
                checked={false}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="Friends"
              />
              <span
                class="checkmark"
              />
            </label>
            <label
              class="container"
            >
              Work
              <input
                checked={false}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="Work"
              />
              <span
                class="checkmark"
              />
            </label>
          </div>
          <input
            className="btn"
            type="submit"
            value="Add Contact"
          />
        </form>
      </div>
      <div>
        <table>
          <tbody>
            <tr>
              <th>
                contact type 
              </th>
              <th>
                 Saved
              </th>
            </tr>
            <tr>
              <th>
                family
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Friends
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Work
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Total
              </th>
              <td>
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <h3
      className="no-guest"
    >
      Loading ...
    </h3>
  </div>
</div>
`;

exports[`Snapshot test 1`] = `
<div>
  <div
    className="navbar"
  >
    <div
      className="logo"
    >
      <h1>
        CONTACTS BOOK 
      </h1>
      <p>
        Consider your contacts saved 
        <span>
          ❤
        </span>
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <span
        className="sm-hide"
      >
        |
      </span>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </div>
  <div
    className="app-container"
  >
    <div
      className="main"
    >
      <div
        className="filter"
      >
        <div
          className="toggle"
        >
          <label
            class="switch"
          >
            <input
              onChange={[Function]}
              type="checkbox"
            />
            <span
              class="slider round"
            />
          </label>
          <p
            className="lead"
          >
            Show favourite Contacts!
          </p>
        </div>
        <div>
          <input
            className="search"
            onChange={[Function]}
            placeholder="Search Contact by name..."
            type="text"
          />
          <i
            className="fas fa-search search-icon"
          />
        </div>
      </div>
      <div
        className="invite-section"
      >
        <h1>
          Create Contact
        </h1>
        <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={true}
            type="text"
            value=""
          />
          <input
            name="phone"
            onChange={[Function]}
            placeholder="Phone"
            required={true}
            type="text"
            value=""
          />
          <p
            className="options-label"
          >
            Contact Type
          </p>
          <div
            className="options"
          >
            <label
              class="container"
            >
              family
              <input
                checked={true}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="family"
              />
              <span
                class="checkmark"
              />
            </label>
            <label
              class="container"
            >
              Friends
              <input
                checked={false}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="Friends"
              />
              <span
                class="checkmark"
              />
            </label>
            <label
              class="container"
            >
              Work
              <input
                checked={false}
                name="diet"
                onChange={[Function]}
                type="radio"
                value="Work"
              />
              <span
                class="checkmark"
              />
            </label>
          </div>
          <input
            className="btn"
            type="submit"
            value="Add Contact"
          />
        </form>
      </div>
      <div>
        <table>
          <tbody>
            <tr>
              <th>
                contact type 
              </th>
              <th>
                 Saved
              </th>
            </tr>
            <tr>
              <th>
                family
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Friends
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Work
              </th>
              <td>
                0
              </td>
            </tr>
            <tr>
              <th>
                Total
              </th>
              <td>
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <h3
      className="no-guest"
    >
      Loading ...
    </h3>
  </div>
</div>
`;
